tests/convergence/BAE.cpp

```diff
@@ -0,0 +1,383 @@
+//
+// Compile with: make BAE
+//
+// Sample runs:  mpirun -np 4 BAE -m ../../data/inline-segment.mesh -sr 1 -pr 4 -prob 0 -o 1
+//               mpirun -np 4 BAE -m ../../data/inline-quad.mesh -sr 1 -pr 3 -prob 0 -o 2
+//               mpirun -np 4 BAE -m ../../data/inline-quad.mesh -sr 1 -pr 3 -prob 1 -o 2
+//               mpirun -np 4 BAE -m ../../data/inline-quad.mesh -sr 1 -pr 3 -prob 2 -o 2
+//               mpirun -np 4 BAE -m ../../data/inline-tri.mesh -sr 1 -pr 3 -prob 2 -o 3
+//               mpirun -np 4 BAE -m ../../data/star.mesh -sr 1 -pr 2 -prob 1 -o 4 
+//               mpirun -np 4 BAE -m ../../data/fichera.mesh -sr 1 -pr 2 -prob 2 -o 2 
+//               mpirun -np 4 BAE -m ../../data/inline-wedge.mesh -sr 0 -pr 2 -prob 0 -o 2
+//               mpirun -np 4 BAE -m ../../data/inline-hex.mesh -sr 0 -pr 1 -prob 1 -o 3 
+//               mpirun -np 4 BAE -m ../../data/square-disc.mesh -sr 1 -pr 2 -prob 1 -o 2
+//
+// Description:  This example code is used for testing the LF-integrators
+//               (Q,grad v), (Q,curl V), (Q, div v)
+//               by solving the appropriate energy projection problems
+//
+//               prob 0: (grad u, grad v) + (u,v) = (grad u_exact, grad v) + (u_exact, v)
+//               prob 1: (curl u, curl v) + (u,v) = (curl u_exact, curl v) + (u_exact, v)
+//               prob 2: (div  u, div  v) + (u,v) = (div  u_exact, div  v) + (u_exact, v)
+
+#include "mfem.hpp"
+#include <fstream>
+#include <iostream>
+
+using namespace std;
+using namespace mfem;
+
+// H1
+double u_exact(const Vector &x);
+void gradu_exact(const Vector &x, Vector &gradu);
+
+// Vector FE
+void U_exact(const Vector &x, Vector & U);
+// H(curl)
+void curlU_exact(const Vector &x, Vector &curlU);
+double curlU2D_exact(const Vector &x);
+// H(div)
+double divU_exact(const Vector &x);
+
+int dim;
+int prob=0;
+Vector alpha;
+
+int main(int argc, char *argv[])
+{
+   // 1. Initialize MPI.
+   int num_procs, myid;
+   MPI_Init(&argc, &argv);
+   MPI_Comm_size(MPI_COMM_WORLD, &num_procs);
+   MPI_Comm_rank(MPI_COMM_WORLD, &myid);
+
+   // 2. Parse command-line options.
+   const char *mesh_file = "../data/inline-quad.mesh";
```

Comment:
Needs another "../" to find the default mesh file.

Reply:
done